src/Servers/Kestrel/perf/Microbenchmarks/NamedPipesTransportBenchmark.cs

```diff
@@ -0,0 +1,163 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO.Pipes;
+using System.Security.Principal;
+using System.Text;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Microbenchmarks;
+
+public class NamedPipesTransportBenchmark
+{
+    private const int _parallelCount = 10;
+    private const int _parallelCallCount = 1000;
+    private const string _pipeName = "MicrobenchmarksTestPipe";
+    private const string _plaintextExpectedResponse =
+        "HTTP/1.1 200 OK\r\n" +
+        "Content-Length: 13\r\n" +
+        "Date: Fri, 02 Mar 2018 18:37:05 GMT\r\n" +
+        "Content-Type: text/plain\r\n" +
+        "Server: Kestrel\r\n" +
+        "\r\n" +
+        "Hello, World!";
+    private static readonly byte[] _responseBuffer = new byte[_plaintextExpectedResponse.Length];
+
+    private IHost _host;
+
+    [Params(1, 2, 8, 16)]
+    public int ListenerQueueCount { get; set; }
+
+    [GlobalSetup]
+    public void GlobalSetupPlaintext()
+    {
+        _host = new HostBuilder()
+            .ConfigureWebHost(webHostBuilder =>
+            {
+                webHostBuilder
+                    // Prevent VS from attaching to hosting startup which could impact results
+                    .UseSetting("preventHostingStartup", "true")
+                    .UseKestrel()
+                    // Bind to a single non-HTTPS endpoint
+                    .UseUrls($"http://pipe:/{_pipeName}")
+                    .Configure(app => app.UseMiddleware<PlaintextMiddleware>())
+                    .UseNamedPipes(options =>
```

Comment:
This is a windows only API, but somehow our microbenchmark validation still runs? That makes me think it's either not working or skipping a benchmark that fails to run? Could you check the behavior here?

Reply:
Weird. I've added an attribute to filter out this benchmark outside of Windows.

New PR for further testing: https://github.com/dotnet/aspnetcore/pull/46491

Reply:
The reason it passed is validation only works on Windows: https://github.com/dotnet/aspnetcore/issues/46495